test/words/test/TestINodeRepeat.java

```diff
@@ -32,4 +33,25 @@ public void testBadRepeat() throws WordsRuntimeException {
 		AST iterativeLoop = new INodeRepeat(stringLeaf, statementList);
 		iterativeLoop.eval(environment);
 	}
+	
+	@Test
+	public void canRepeatObjectCreation() {
+		AST statementList = new INodeStatementList(createObjectFred);
+		AST iterativeLoop = new INodeRepeat(fiveLeaf, statementList);
+		loop.enqueueAST(iterativeLoop);
+		loop.fastForwardEnvironment(5);
+		assertEquals("Successfully created multiple Freds", environment.getObjects().size(), 5);
+		assertEquals("Outside of loop, ends in global scope", environment.getNumberOfScopes(), 1);
+	}
+	
+	@Test
+	public void scopeHandlesExceptions() throws WordsRuntimeException {
+		AST statementList = new INodeStatementList(createObjectFred);
+		AST iterativeLoop = new INodeRepeat(fiveLeaf, statementList);
+		loop.enqueueAST(createObjectFred);
+		loop.fastForwardEnvironment(1);
+		// this iterative loop will fail 5 times
+		loop.enqueueAST(iterativeLoop);
```

Comment:
Are you sure this test is working?  It looks to me like you're not fast forwarding the environment to actually run the loop, so the exceptions of trying to create Fred when he already exists in outer scope are never actually generated